CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -651,6 +659,33 @@ private static void InlinesToHtml(HtmlTextWriter writer, Inline inline, CommonMa
             }
         }
 
+        private static void WriteGitHubHeadingIds(HtmlTextWriter writer, Inline inline, CommonMarkSettings settings, Stack<InlineStackEntry> stack) {
+            using (var tempWriter = new System.IO.StringWriter()) {
+                var tempWrapper = new HtmlTextWriter(tempWriter);
+                var tempStringBuilder = new StringBuilder();
+                InlinesToPlainText(tempWrapper, inline, stack);
+                string plaintextContent = tempWriter.ToString();
+
+                // Normalize plaintext content according to GitHub ID rules
+                plaintextContent = new Regex(@"[^\w\-\ ]").Replace(plaintextContent, "").Replace(" ", "-");
```

Comment:
you can get the unicode category of a char. if you think regex is better, you will have to show it in a benchmark. also your approach to getting lowercase ascii letters is rather limited, e.g. it does not handle unicode chars. and it can be replaced by dictionary comparer.